src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ParameterTest/TvpTest.cs

```diff
@@ -82,6 +83,97 @@ public void TestPacketNumberWraparound()
             Assert.True(enumerator.MaxCount == enumerator.Count);
         }
 
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup), nameof(DataTestUtility.IsNotAzureServer))]
+        public void TestConnectionIsSafeToReuse()
+        {
+            using SqlConnection connection = new(DataTestUtility.TCPConnectionString);
+
+            // Bad Scenario - exception expected.
+            try
+            {
+                List<A> list = new()
+                {
+                    new A(0),
+                    null,
+                    new A(2),
+                    new A(3),
+                    new A(4),
+                    new A(5)
+                };
+
+                IEnumerable<int> Ids = list.Select(x => x.id.Value).Distinct();
+
+                var sqlParam = new SqlParameter("ids", SqlDbType.Structured)
+                {
+                    TypeName = "dbo.TableOfIntId",
+                    SqlValue = Ids.Select(x =>
+                    {
+                        SqlDataRecord rec = new(new[] { new SqlMetaData("Id", SqlDbType.Int) });
+                        rec.SetInt32(0, x);
+                        return rec;
+                    })
+                };
+
+                var parameters = new List<SqlParameter>() { sqlParam };
+                const string SQL = @"SELECT * FROM information_schema.COLUMNS cols INNER JOIN  @ids Ids on Ids.id = cols.ORDINAL_POSITION";
+                using SqlCommand cmd = new(SQL, connection);
+                cmd.CommandTimeout = 100;
+                AddCommandParameters(cmd, parameters);
+                new SqlDataAdapter(cmd).Fill(new("BadFunc"));
+            }
+            catch
+            {
+                // Ignore this exception as it's deliberately introduced.
```

Comment:
It might be worth asserting that the expected exception did occur so it can't be broken in some other way in future.